### PR TITLE
Support array sum aggregation function

### PR DESCRIPTION
**Pull request by xiangfu0**
- Support Aggregation function `SumArrayLong` and `SumArrayDouble` to aggregate on Arrays to sum on the corresponding array positions.

Sample queries:

```
SELECT sumArrayLong(longArrayCol), sumArrayDouble(doubleArrayCol) FROM myTable
```

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/operator/docvalsets/RowBasedBlockValSet.java [hunk 1]
```diff
@@ -400,27 +400,212 @@ public int[][] getDictionaryIdsMV() {
 
   @Override
   public int[][] getIntValuesMV() {
-    throw new UnsupportedOperationException();
+    int numRows = _rows.size();
+    int[][] values = new int[numRows][];
+    if (numRows == 0 || _dataType == DataType.UNKNOWN) {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/docvalsets/RowBasedBlockValSet.java:405]: Why skipping `UNKNOWN`? Should we fill default value when it is unknown?

File: pinot-core/src/main/java/org/apache/pinot/core/operator/docvalsets/RowBasedBlockValSet.java [hunk 2]
```diff
@@ -400,27 +400,212 @@ public int[][] getDictionaryIdsMV() {
...
+      return values;
+    }
+    for (int i = 0; i < numRows; i++) {
+      Object storedValue = _rows.get(i)[_colId];
+      if (storedValue instanceof int[]) {
+        values[i] = (int[]) storedValue;
+      } else if (storedValue instanceof long[]) {
+        long[] longArray = (long[]) storedValue;
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/docvalsets/RowBasedBlockValSet.java:413]: You may use `ArrayCopyUtils` for the conversions

File: pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/array/SumArrayDoubleAggregationFunction.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+import org.apache.pinot.core.query.aggregation.AggregationResultHolder;
+import org.apache.pinot.core.query.aggregation.ObjectAggregationResultHolder;
+import org.apache.pinot.core.query.aggregation.function.BaseSingleInputAggregationFunction;
+import org.apache.pinot.core.query.aggregation.groupby.GroupByResultHolder;
+import org.apache.pinot.core.query.aggregation.groupby.ObjectGroupByResultHolder;
+import org.apache.pinot.segment.spi.AggregationFunctionType;
+
+
+public class SumArrayDoubleAggregationFunction
+    extends BaseSingleInputAggregationFunction<double[], DoubleArrayList> {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/array/SumArrayDoubleAggregationFunction.java:36]: I don't think `double[]` can be used as intermediate result. We do support `DoubleArrayList`

Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/array/SumArrayDoubleAggregationFunction.java:36]: Please add a v1 test to ensure it works as expected

Comment (xiangfu0) [reply pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/array/SumArrayDoubleAggregationFunction.java]: Added.

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**